Automattic/woocommerce-payments
# Tweak: Refactor subscription traits

Fixes #2385 

#### Changes proposed in this Pull Request


# Making Subscription Traits More Reusable

This is a simple code tweak to allow certain subscription trait functions to be more reusable. This is particularly useful for the `UPE_Payment_Method` class, which previously copied code to check whether subscriptions were enabled: now this class can simply import the functions from the new subscriptions utility trait, which is also imported by the main subscriptions trait. 




#### Testing instructions


1. Ensure you have pulled the changes in this PR, the latest version of the Payments Server, and that you have the WC Subscriptions plugin enabled in your site. Also, please enable the UPE gateway using the WCPay Dev Plugin or simply set `_wcpay_feature_upe` to a value of 1 in your `wp_options` table.
2. There isn't any new functionality here, so just complete a checkout with a subscription product, run the tests, and ensure that everything works as expected. 
3. Go to your happy place.


-------------------
